### PR TITLE
add ranger filters

### PR DESCRIPTION
**Pull request by bommert**
Impurity and permutation importance filter based on ranger

### CODE REVIEW

File: R/Filter.R [hunk 1]
```diff
@@ -631,3 +631,46 @@ makeFilter(
...
+#' @rdname makeFilter
+#' @name makeFilter
+makeFilter(
+  name = "ranger.permutation",
+  desc = "Variable importance based on ranger permutation importance",
+  pkg  = "ranger",
+  supported.tasks = c("classif", "regr", "surv"),
+  supported.features = c("numerics", "factors", "ordered"),
+  fun = function(task, nselect, ...) {
+    lrn.type = paste0(getTaskType(task), ".ranger")
```
Comment (jakob-r) [inline R/Filter.R]: Maybe `ranger.permutation` because we have at least two?

File: R/Filter.R [hunk 2]
```diff
@@ -631,3 +631,46 @@ makeFilter(
...
+#' @rdname makeFilter
+#' @name makeFilter
+makeFilter(
+  name = "ranger.impurity",
+  desc = "Variable importance based on ranger impurity importance",
+  pkg  = "ranger",
+  supported.tasks = c("classif", "regr"),
+  supported.features = c("numerics", "factors", "ordered"),
+  fun = function(task, nselect, ...) {
+    lrn.type = paste0(getTaskType(task), ".ranger")
```
Comment (jakob-r) [inline R/Filter.R]: see above.

File: R/Filter.R (hunk 1 above)
Comment (bommert) [reply R/Filter.R]: I'm confused. The name of the filter is ranger.permutation already. What exactly do you want me to change/ reconsider?

**Comment by bommert** [comment]
In test_featsel_filters.R all filters are tested automatically.

File: R/Filter.R (hunk 1 above)
Comment (jakob-r) [reply R/Filter.R]: My bad...

**Comment by jakob-r** [comment]
On a third thought... I don't like this so much because it is redundant with `getFeatureImportanceLearner.*` We should have a filter that can use all implemented `getFeatureImportanceLearner`functions. 

But as we already have cforest.importance, randomForestSRC.var.select, randomForestSRC.rfsrc, and randomForest.importance it seems that the horse is already out the barn.

Nonetheless a general solution would be great if you are interested ;)

**Comment by mllg** [comment]
I second what @jakob-r said, a feature importance filter where you pass a learner with the respective property would be useful. Nevertheless, this one here is also fine.

**Comment by larskotthoff** [comment]
This appears to have broken FDA?

**Comment by mllg** [comment]
That's odd. These things have nothing to do with each other. I'll look into it tomorrow.

**Comment by larskotthoff** [comment]
Could be unrelated, but Travis started failing after this was merged.

**Comment by mllg** [comment]
Update of package `fda.usc` to version 1.4.0 caused this, I'm opening a new issue.

**Pull request merged**